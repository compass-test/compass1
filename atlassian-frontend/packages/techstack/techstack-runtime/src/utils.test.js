const { getChecksForTeckStack } = require('./utils');

const createCheck = (type, ruleName) => ({
  type,
  plugin: 'plugin',
  rule: ruleName,
  configuration: 'error',
});

const createSolution = key => {
  return {
    id: `solution-${key}`,
    checks: () => [
      createCheck('eslint', `check-eslint-${key}`),
      createCheck('stricter', `check-stricter-${key}`),
    ],
    antiChecks: () => [
      createCheck('eslint', `anti-check-eslint-${key}`),
      createCheck('stricter', `anti-check-stricter-${key}`),
    ],
  };
};

const createSolutionWithoutChecks = key => {
  return {
    id: `solution-${key}`,
  };
};

const techstackDefinition = [
  {
    id: 'use-case-a',
    solutions: [
      createSolution('a-a'),
      createSolution('a-b'),
      createSolution('a-c'),
      createSolutionWithoutChecks('a-d'),
    ],
  },
  {
    id: 'use-case-b',
    solutions: [
      createSolution('b-a'),
      createSolution('b-b'),
      createSolution('b-c'),
      createSolutionWithoutChecks('b-d'),
    ],
  },
];

const techstack = { 'use-case-a': ['solution-a-a'] };
const techstack2 = { 'use-case-b': ['solution-b-a', 'solution-b-b'] };

describe('Utils for techstack tests', () => {
  test('getChecksForTechStack should return correct checks', () => {
    // dependency on an external package, need to mock it properly instead
    // but okay for now
    expect(
      getChecksForTeckStack({
        techstackDefinition,
        techstack,
        config: {},
      }),
    ).toMatchSnapshot();

    expect(
      getChecksForTeckStack({
        techstackDefinition,
        techstack: techstack2,
        config: {},
      }),
    ).toMatchSnapshot();
  });
});

module.exports = {
  techstackDefinition,
  techstack,
  techstack2,
};
