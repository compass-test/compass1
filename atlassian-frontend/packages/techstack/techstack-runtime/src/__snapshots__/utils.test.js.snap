// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils for techstack tests getChecksForTechStack should return correct checks 1`] = `
Array [
  Object {
    "configuration": "error",
    "plugin": "plugin",
    "rule": "check-eslint-a-a",
    "type": "eslint",
  },
  Object {
    "configuration": "error",
    "plugin": "plugin",
    "rule": "check-stricter-a-a",
    "type": "stricter",
  },
  Object {
    "configuration": "error",
    "plugin": "plugin",
    "rule": "anti-check-eslint-a-b",
    "type": "eslint",
  },
  Object {
    "configuration": "error",
    "plugin": "plugin",
    "rule": "anti-check-stricter-a-b",
    "type": "stricter",
  },
  Object {
    "configuration": "error",
    "plugin": "plugin",
    "rule": "anti-check-eslint-a-c",
    "type": "eslint",
  },
  Object {
    "configuration": "error",
    "plugin": "plugin",
    "rule": "anti-check-stricter-a-c",
    "type": "stricter",
  },
  Object {
    "configuration": "error",
    "plugin": "plugin",
    "rule": "anti-check-eslint-b-a",
    "type": "eslint",
  },
  Object {
    "configuration": "error",
    "plugin": "plugin",
    "rule": "anti-check-stricter-b-a",
    "type": "stricter",
  },
  Object {
    "configuration": "error",
    "plugin": "plugin",
    "rule": "anti-check-eslint-b-b",
    "type": "eslint",
  },
  Object {
    "configuration": "error",
    "plugin": "plugin",
    "rule": "anti-check-stricter-b-b",
    "type": "stricter",
  },
  Object {
    "configuration": "error",
    "plugin": "plugin",
    "rule": "anti-check-eslint-b-c",
    "type": "eslint",
  },
  Object {
    "configuration": "error",
    "plugin": "plugin",
    "rule": "anti-check-stricter-b-c",
    "type": "stricter",
  },
]
`;

exports[`Utils for techstack tests getChecksForTechStack should return correct checks 2`] = `
Array [
  Object {
    "configuration": "error",
    "plugin": "plugin",
    "rule": "anti-check-eslint-a-a",
    "type": "eslint",
  },
  Object {
    "configuration": "error",
    "plugin": "plugin",
    "rule": "anti-check-stricter-a-a",
    "type": "stricter",
  },
  Object {
    "configuration": "error",
    "plugin": "plugin",
    "rule": "anti-check-eslint-a-b",
    "type": "eslint",
  },
  Object {
    "configuration": "error",
    "plugin": "plugin",
    "rule": "anti-check-stricter-a-b",
    "type": "stricter",
  },
  Object {
    "configuration": "error",
    "plugin": "plugin",
    "rule": "anti-check-eslint-a-c",
    "type": "eslint",
  },
  Object {
    "configuration": "error",
    "plugin": "plugin",
    "rule": "anti-check-stricter-a-c",
    "type": "stricter",
  },
  Object {
    "configuration": "error",
    "plugin": "plugin",
    "rule": "check-eslint-b-a",
    "type": "eslint",
  },
  Object {
    "configuration": "error",
    "plugin": "plugin",
    "rule": "check-stricter-b-a",
    "type": "stricter",
  },
  Object {
    "configuration": "error",
    "plugin": "plugin",
    "rule": "check-eslint-b-b",
    "type": "eslint",
  },
  Object {
    "configuration": "error",
    "plugin": "plugin",
    "rule": "check-stricter-b-b",
    "type": "stricter",
  },
  Object {
    "configuration": "error",
    "plugin": "plugin",
    "rule": "anti-check-eslint-b-c",
    "type": "eslint",
  },
  Object {
    "configuration": "error",
    "plugin": "plugin",
    "rule": "anti-check-stricter-b-c",
    "type": "stricter",
  },
]
`;
